services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/ExecutionEngineService.java

```diff
@@ -0,0 +1,156 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.services.powchain.execution;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.errorprone.annotations.FormatMethod;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import org.web3j.protocol.core.methods.response.EthBlock.Block;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.logging.ColorConsolePrinter;
+import tech.pegasys.teku.infrastructure.logging.ColorConsolePrinter.Color;
+import tech.pegasys.teku.infrastructure.logging.LogFormatter;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.services.powchain.execution.client.ExecutionEngineClient;
+import tech.pegasys.teku.services.powchain.execution.client.Web3JExecutionEngineClient;
+import tech.pegasys.teku.services.powchain.execution.client.schema.AssembleBlockRequest;
+import tech.pegasys.teku.services.powchain.execution.client.schema.ExecutionPayload;
+import tech.pegasys.teku.services.powchain.execution.client.schema.NewBlockResponse;
+import tech.pegasys.teku.services.powchain.execution.client.schema.Response;
+import tech.pegasys.teku.spec.executionengine.ExecutionEngineChannel;
+
+public class ExecutionEngineService implements ExecutionEngineChannel {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final ExecutionEngineClient executionEngineClient;
+
+  public static ExecutionEngineService create(String eth1EngineEndpoint) {
+    checkNotNull(eth1EngineEndpoint);
+    return new ExecutionEngineService(new Web3JExecutionEngineClient(eth1EngineEndpoint));
+  }
+
+  public static ExecutionEngineService createStub() {
+    return new ExecutionEngineService(ExecutionEngineClient.Stub);
+  }
+
+  public ExecutionEngineService(ExecutionEngineClient executionEngineClient) {
+    this.executionEngineClient = executionEngineClient;
+  }
+
+  private static <K> K unwrapResponseOrThrow(Response<K> response) {
+    checkArgument(
+        response.getPayload() != null, "Invalid remote response: %s", response.getReason());
+    return response.getPayload();
+  }
+
+  @FormatMethod
+  private static void printConsole(String formatString, Object... args) {
+    LOG.info(ColorConsolePrinter.print(String.format(formatString, args), Color.CYAN));
+  }
```

Comment:
This seems useful for debugging but likely becomes trace level in actual production code right?

Reply:
Yep, sure!

Reply:
Just tried to preserve the original code. Would be probably useful for interop